core/src/cluster_info.rs

```diff
@@ -544,27 +511,47 @@ impl ClusterInfo {
         stakes_and_index
     }
 
-    /// Return sorted Retransmit peers and index of `Self.id()` as if it were in that list
-    pub fn shuffle_peers_and_index<S: std::hash::BuildHasher>(
-        &self,
-        stakes: Option<&HashMap<Pubkey, u64, S>>,
+    fn stake_weighted_shuffle(
+        stakes_and_index: &[(u64, usize)],
```

Comment:
Is it guaranteed that two nodes will have `stakes_and_index` in the same order? The reason I ask this, say two index have same stake, the weighted shuffle at line 520 will return the same output on different node. But line 522 may return different index map if `stakes_and_index` could be different on different nodes.

Hope I am making sense.

Reply:
assuming both nodes see the same network state, stakes_and_index are guaranteed to be identical since if the stake is equal we sort it by pubkey. So both nodes should sort it identically. 
After that, since the weighted shuffle uses a seeded rng it should return the same shuffle right?

Reply:
Yes, if `stakes_and_index` are in same order, everything else should line up..

Reply:
yep it's a stable sort :)